### PR TITLE
Fix #36: add stylus support

### LINKED ISSUE #36: Check compatibility in Stylus stylesheets
https://learnboost.github.io/stylus/

### PR DESCRIPTION
**Pull request by Faeranne**
easily add more support in sourceParse.js

### CODE REVIEW

**Comment by mrmakeit** [comment]
@groovecoder @openjck r?

**Comment by mrmakeit** [comment]
@darkwing r?

**Comment by openjck** [comment]
Modified the title so that merging this automatically closes #36.

**Comment by openjck** [comment]
My apologies for the delay. This is an awesome addition. I'm looking now.

File: module.js [hunk 1]
```diff
@@ -25,17 +26,21 @@ var github = function(payload, localToken) {
         var parsedBody = JSON.parse(body);
         var files = parsedBody.files;
         var configMetadataURL = payload.repository.contents_url.replace('{+path}', '.doiuse');
-        request({ url: configMetadataURL, headers: {'User-Agent': 'YouShouldUse'}}, function(err, res, body) {
+        request({
+            url: configMetadataURL,
+            headers: {
+                'User-Agent': 'YouShouldUse'
```
Comment (openjck) [inline module.js]: Nice! Glad we're using the brand name as the user-agent here. :smile:

**Comment by openjck** [comment]
Thanks for running js-beautify and jshint against this. The code looks great!

File: sourceParse.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+'use strict';
```
Comment (openjck) [inline sourceParse.js]: Nice! :smile:

File: sourceParse.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        }
+    }, function(err, res, body) {
+        var contents = body;
+        var stylize = style(contents)
+            .set('filename', file.filename)
+            .set('sourcemap', {});
+        stylize.render(function(err, css) {
+            var source = new sourceMap.SourceMapConsumer(stylize.sourcemap);
+            var onFeatureUsage = function(feature) {
+                var newPos = source.originalPositionFor(feature.usage.source.start);
```
Comment (openjck) [inline sourceParse.js]: Awesome!

Comment (openjck) [inline sourceParse.js]: This is fantastic. Nice attention to detail.

**Comment by openjck** [comment]
David might not be the only one having issues. I'm no longer getting comments, [even to CSS](https://github.com/openjck/YouShouldUse-test/commit/3928f8241c4c6e4bf207be1950630058f50ffe74), so I'm unfortunately unable to test this at the moment.

**Comment by openjck** [comment]
https://github.com/openjck/YouShouldUse-test/commit/3c00548962791904b11d5351e447eb2627b1604e

Beyond epic.

**Pull request merged**